### ISSUE TITLE
Segmentation Fault in resizeFixedAspectRatio (Ubuntu)

### ISSUE THREAD

**Issue by johaq** [CMU-Perceptual-Computing-Lab/openpose#110]
Hi, 
I'm trying to extract poses from a cv image. I looked at the example code, but right now I got the following problem:
I call cvMatToOpInput.format:
```
op::CvMatToOpInput cvMatToOpInput{netInputSize, scaleNumber, (float)scaleGap};
std::tie(netInputArray, scaleRatios) = cvMatToOpInput.format(inputImage);
```
netInputSize: (656,368)
scaleNumber: 1.0
scaleGap: 0.3
inputImage Size: (640,480)

In cvMatToOpInput.format I get a segfault. Using gdb i could track the problem to the resizeFixedAspectRatio method. Backtrace:
```
Thread 1 "detect_people_s" received signal SIGSEGV, Segmentation fault.
0x00007ffff604e901 in op::resizeFixedAspectRatio(cv::Mat const&, double, op::Point<int> const&, int, cv::Scalar_<double> const&) () from /home/johannes/robocup/openpose/build/lib/libopenpose.so.1.0.0-rc3
(gdb) bt
#0  0x00007ffff604e901 in op::resizeFixedAspectRatio(cv::Mat const&, double, op::Point<int> const&, int, cv::Scalar_<double> const&) () from /home/johannes/robocup/openpose/build/lib/libopenpose.so.1.0.0-rc3
#1  0x00007ffff6060806 in op::CvMatToOpInput::format(cv::Mat const&) const () from /home/johannes/robocup/openpose/build/lib/libopenpose.so.1.0.0-rc3
```

Did anyone else experience this issue? Could there be something wrong with my opencv installation? Or is it a simple config error and I missed the guide on how to set scalenum and scalegap correctly?

### Type of issue
You might select multiple topics, delete the rest:
- Execution error
- Help wanted
- Question

### Your system configuration
**Operating system** : Ubuntu 16.04
**CUDA version** : 8.0
**cuDNN version**: 6.0
**GPU model** (`nvidia-smi`): GeForce 940MX
**Caffe version**: Default from OpenPose
**OpenCV version**: 3.2
Compiler: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609

**Comment by johaq** [comment]
I solved my problem. I apparently had some issues with conflicting opencv versions (2.4 and 3.2).
If someone has a similar problem: What I did was sudo apt purging opencv2 and then a clean install of openpose and caffee. There is probably a less severe solution but this did it for me.

**Comment by gineshidalgo99** [comment]
Thank you for the feed-back.
Having 2 working OpenCV is quite tricky, your solution is actually the easiest one.

**Comment by lisayan** [comment]
Can you link to how you purged opencv completely and reinstalled?

**Comment by johaq** [comment]
well for me it was
```
sudo apt purge --remove libopencv-core2.4v5
```
But your installation might not be from that package.
The installation of opencv that is currently working for me is installed from source with contributions and cuda. I would also recommend rechecking the Makefile.config files in caffe and openpose to match the opencv version you want to use.

**Comment by gineshidalgo99** [comment]
I'd say `sudo apt purge --remove libopencv && sudo apt purge --remove libopencv*` might be more generic (but I did not try it since I cannot remove them without messing some of my programs that use them)

**Comment by lisayan** [comment]
I found that I had to uncomment a line in Makefile.config to set OpenCV version to 3 in openpose and also openpose/3rdparty/caffe, but it seems that I'm still running into the same issue. Thanks for the responses though! I'll keep looking into it

**Comment by gineshidalgo99** [comment]
Usually the fastest option is to only use OpenCV 2.4 (the one from `sudo apt-get install libopencv`). That one always work (to the best of my knowledge).

I've actually just read on some StackOverflow post that OpenCV 3.2 does not work with Caffe, that 3.1 must be used instead. Not 100% sure whether this is true but it might be the case.

**Comment by johaq** [comment]
Cannot confirm. I am building tag 3.2.0-rc of opencv and it works fine for me.

**Comment by gineshidalgo99** [comment]
@johaq Thanks for clarifying it!

**Comment by Omegastick** [comment]
I've got this same problem. I've tried it on two machines (both Lenovo Thinkstation P320 Tinies), one of which had a completely fresh install of Ubuntu. I've tried the fixes listed here on both machines, including removing all versions of OpenCV and only installing the default (libopencv-dev) as well as building both 3.2.0 and 3.4.1 myself. I've also tried it with ROS's OpenCV 3. 

It seems to be segfaulting at that resizeFixedAspectRatio line.
Any ideas what I can do to fix it?
